bundles/org.openhab.binding.senseenergy/src/main/java/org/openhab/binding/senseenergy/internal/handler/SenseEnergyMonitorHandler.java

```diff
@@ -204,7 +204,7 @@ public void heartbeat() {
         ThingStatus thingStatus = getThing().getStatus();
 
         if (thingStatus == ThingStatus.OFFLINE
-                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.COMMUNICATION_ERROR) {
+                && getThing().getStatusInfo().getStatusDetail() != ThingStatusDetail.CONFIGURATION_ERROR) {
```

Comment:
Does this change matter now that `handleApiException` is changed?
What other statusdetails would the heartbeat see here other then `COMMUNICATION_ERROR` that it should call `goOnline()` ?

Reply:
For the specific issue I was seeing, it would not matter, however, I think this is more robust for unforeseen Exceptions.  My thinking is that if there is any exception where it goes offline (with the exception of a CONFIGURATION_ERROR), it should try to recover.

Reply:
But all exceptions by `handleApiExceprion` are now set to communication_error.

Without manually checking all code paths, the previous check seems better.